torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2491,6 +2491,42 @@ Tensor linalg_eig_backward(const std::vector<torch::autograd::Variable> &grads,
   }
 }
 
+// https://people.maths.ox.ac.uk/gilesm/files/NA-08-01.pdf, page 10
+// see also https://arxiv.org/pdf/1701.00392.pdf Eqs. (4.60) and (4.63)
+Tensor linalg_eig_jvp_eigenvalues(const Tensor& dA,
+                                  const Tensor& L,
+                                  const Tensor& V) {
+  const auto dAComplex = at::complex(dA, at::zeros(dA.size(-1), dA.options()));
+  const auto dVfactor = at::linalg_solve(V, at::matmul(dAComplex, V));
+  return at::diag(at::mul(at::eye(V.size(-1), V.options()), dVfactor));
+}
+
+Tensor linalg_eig_jvp_eigenvectors(const Tensor& dA,
+                                   const Tensor& L,
+                                   const Tensor& V) {
+  const auto dAComplex = at::complex(dA, at::zeros(dA.size(-1), dA.options()));
```

Comment:
A better way of doing this might be the following
```suggestion
  const auto dAComplex = dA.to(c10::toComplexType(dA.scalar_type()));
```
Also, doesn't the current way assume that `dA` is real? This may not be the case if the input matrix is complex. In any case, the proposed change handles this case.

Same in the other places.

Reply:
Good catch :-) I used a real matrix in my example, so I would not notice if it only worked for real dA

Reply:
Weird that the tests didn't catch this one... cc @albanD

Reply:
Nevermind, the tests are red, so it could potentially be caught by the tests only that those reds are shadowing this.

On another note, @anjali411 it would be cool to have a shortcut to do this, casting a real matrix to its complex counterpart taking care of whether it's half / single / double precision.

Reply:
BTW, I still get the error from test_forward_mode_AD_corrcoef_cpu_complex128, test_forward_mode_AD_corrcoef_cpu_float64. I did not work on the correlation coefficients, so is this unrelated or could it be a side effect?

Reply:
I think that `corrcoef` might be a composite function of this function, so it may so happen that you need to activate forward AD for these function. Just add the pertinent flag to their `OpInfo`.

Reply:
Added the `supports_forward_ad` flag, now it is working :+1: